src/analyze/analyze.c

```diff
@@ -82,6 +85,40 @@ static const char *arg_root = NULL;
 STATIC_DESTRUCTOR_REGISTER(arg_dot_from_patterns, strv_freep);
 STATIC_DESTRUCTOR_REGISTER(arg_dot_to_patterns, strv_freep);
 
+#define DOT_COLOR_AFTER     "green"
+#define DOT_COLOR_REQUIRES  "black"
+#define DOT_COLOR_REQUISITE "blue"
+#define DOT_COLOR_WANTS     "grey66"
+#define DOT_COLOR_CONFLICTS "red"
+#define DOT_COLOR_BINDSTO   "magenta"
+#define DOT_COLOR_ONFAILURE "orange"
+#define DOT_COLOR_TRIGGERS  "grey66"
+#define DOT_STYLE_AFTER     "solid"
+#define DOT_STYLE_REQUIRES  "solid"
+#define DOT_STYLE_REQUISITE "solid"
+#define DOT_STYLE_WANTS     "solid"
+#define DOT_STYLE_CONFLICTS "solid"
+#define DOT_STYLE_BINDSTO   "solid"
+#define DOT_STYLE_ONFAILURE "solid"
+#define DOT_STYLE_TRIGGERS  "dotted"
+
+static char *color_after     = NULL;
+static char *color_requires  = NULL;
+static char *color_requisite = NULL;
+static char *color_wants     = NULL;
+static char *color_conflicts = NULL;
+static char *color_bindsto   = NULL;
+static char *color_onfailure = NULL;
+static char *color_triggers  = NULL;
+static char *style_after     = NULL;
+static char *style_requires  = NULL;
+static char *style_requisite = NULL;
+static char *style_wants     = NULL;
+static char *style_conflicts = NULL;
+static char *style_bindsto   = NULL;
+static char *style_onfailure = NULL;
+static char *style_triggers  = NULL;
```

Comment:
instead of having so many separate variables for this. Maybe it would be nicer to simply make this a single array:

```c
static char *arg_dependency_style[_UNIT_DEPENDENCY_MAX];
```

and then reference the style you need with `arg_dependency_style[UNIT_DEPENDENCY_AFTER]` and so on. 

This would allocate slightly too much memory (as there are more deps defined then you want fields here), but that shouldn't really matter, given it's small.

It also allows us to free these fields in one go by iterating through every field in the enum

Reply:
Created two arrays dependency_color and dependency_style.